demo/rags_compare.py

```diff
@@ -0,0 +1,489 @@
+            self.inited[retriever] = False
+            self.cnt[retriever] = 10
+            self.add_retriever(
+                retriever, self.init_Ensemble_Retriever(context=context)
+            )
+            self.inited[retriever] = True
+            pass
+        elif retriever == 'MultiVector_Retriever' and retriever not in self.inited:
+            self.inited[retriever] = False
+            self.cnt[retriever] = 10
+            self.add_retriever(
+                retriever, self.init_MultiVector_Retriever(context=context)
+            )
+            self.inited[retriever] = True
+            pass
+        elif retriever == 'Parent_Document_Retriever' and retriever not in self.inited:
+            self.inited[retriever] = False
+            self.cnt[retriever] = 10
+            self.add_retriever(
+                retriever,
+                self.init_Parent_Document_Retriever(context=context)
+            )
+            self.inited[retriever] = True
+            pass
+        elif retriever == 'llamaindex' and retriever not in self.inited:
+            self.inited[retriever] = False
+            self.cnt[retriever] = 10
+            documents = []
+            if not self.pdf_path == None:
+                documents = SimpleDirectoryReader(input_files=[self.pdf_path]
+                                                  ).load_data()
+            elif not context == None:
+                name = 'llamaindex'
+                self.data[name] = context
+                self.timer[name] = time.time()
+                cur_time = self.timer[name]
+                while (cur_time - self.timer[name] < self.WaitTime):
+                    time.sleep(1)
+                    cur_time = time.time()
+                documents = self.data[name]
+            index = VectorStoreIndex.from_documents(documents)
+            llamaindex = index.as_retriever()
+            self.add_retriever(retriever, llamaindex)
+            self.inited[retriever] = True
+            pass
+
+
+def langchain_docs_to_string(docs: List) -> str:
+    return f"---".join([
+        f"Document {i+1}:\n\n" + d.page_content for i, d in enumerate(docs)
+    ])
+
+
+def llamaindex_docs_to_string(docs: List) -> str:
+    return f"---".join([
+        f"Document {i+1}:\n\n" + d.text for i, d in enumerate(docs)
+    ])
+
+
+r_m_ps = retriever_model_prompt()
+
+
+def retriever_method(
+    input: str, retriever: str, context: Optional[Dict] = None
+) -> str:
+    res = ""
+    global r_m_ps
+
+    print(retriever, input)
+    if not context == None:
+        if retriever == 'llamaindex':
+            context = deal_llamaindex(context=context)
+        else:
+            context = deal_context(context=context)
+    r_m_ps.init_retriever(retriever=retriever, context=context)
+
+    if retriever == 'MultiQueryRetriever':
+        flag = 1
+        while r_m_ps.inited[retriever] == False:
+            print('line:302')
```

Comment:
clean debug code here(print XXX) ? or make it more clear

Reply:
ok